test/remote-test.js

```diff
@@ -1873,3 +1960,499 @@ describe('Remote', function() {
+      ],
+      optional: [
+        ['binary', false]
+      ],
+      default: [
+        ['binary', true]
+      ]
+    }),
+    makeRequest('transaction_entry', 'requestTransactionEntry', {
+      required: [
+        ['hash', [['tx_hash', TRANSACTION_HASH]]]
+      ],
+      optional: [
+        REQOPTION.LEDGER,
+      ],
+      default: [
+        ['ledger', [['ledger_index', 'validated']]]
+      ]
+    }),
+
+    makeRequest('tx_history', 'requestTransactionHistory', {
+      optional: [
+        ['start', 10]
+      ],
+      noPositional: true
+    }),
+    makeRequest('book_offers', 'requestBookOffers', {
+      required: [
+        ['gets', [['taker_gets', {currency: HEX_USD, issuer: ADDRESS}]]],
+        ['pays', [['taker_pays', {currency: HEX_USD, issuer: ADDRESS}]]]
+      ],
+      optional: [
+        ['taker', ADDRESS],
+        REQOPTION.LEDGER,
+        ['limit', 10],
+      ],
+      default: [
+        ['taker', ripple.UInt160.ACCOUNT_ONE]
+      ],
+      noPositional: true
+    }),
+
+    makeRequest('ledger', 'requestLedger', {
+      optional: [
+        REQOPTION.LEDGER,
+        ['full', true],
+        ['expand', true],
+        ['transactions', true],
+        ['accounts', true]
+      ],
+      noPositional: true
+    }),
+    makeRequest('ledger_data', 'requestLedgerData', {
+      optional: [
+        REQOPTION.LEDGER,
+        ['binary', false],
+        ['limit', 10]
+      ],
+      default: [
+        ['binary', true]
+      ],
+      noPositional: true
+    }),
+    makeRequest('ledger_entry', 'requestLedgerEntry', {
+      required: [
+        ['type', 'account_root']
+      ],
+      noKeyed: true
+    }),
+    makeRequest('ledger_closed', 'requestLedgerClosed'),
+    makeRequest('ledger_current', 'requestLedgerCurrent'),
+    makeRequest('ledger_header', 'requestLedgerHeader'),
+
+    makeRequest('ledger_entry', 'requestAccountBalance', {
+      alias: 'account_balance',
+      required: [
+        REQOPTION.ACCOUNT_ROOT,
+      ],
+      optional: [
+        REQOPTION.LEDGER,
+      ]
+    }),
+    makeRequest('ledger_entry', 'requestAccountFlags', {
+      alias: 'account_flags',
+      required: [
+        REQOPTION.ACCOUNT_ROOT,
+      ],
+      optional: [
+        REQOPTION.LEDGER,
+      ],
+    }),
+    makeRequest('ledger_entry', 'requestOwnerCount', {
+      alias: 'owner_count',
+      required: [
+        REQOPTION.ACCOUNT_ROOT,
+      ],
+      optional: [
+        REQOPTION.LEDGER,
+      ],
+    }),
+
+    makeRequest('sign', 'requestSign', {
+      required: [
+        ['secret', SECRET],
+        ['tx_json', TX_JSON]
+      ]
+    }),
+    makeRequest('submit', 'requestSubmit')
+  ];
+
+  function checkRequest(request, expectedReqParams) {
+    assert(request.requested, 'Request unattempted, most likely callback was ignored');
+    [['id', undefined], ...expectedReqParams].forEach(arg => {
+      assert.deepEqual(
+        request.message[arg[0]], arg[1],
```

Comment:
I `skipped the skip` and ran these tests locally and it seems there's some tests where `arg[0]` is not a string (key), but actually a [key, value] list it self.